internal/pkg/admission/policy-server-secret.go

```diff
@@ -22,8 +28,66 @@ func (r *Reconciler) reconcileSecret(ctx context.Context, secret *corev1.Secret)
 	return fmt.Errorf("error reconciling policy-server Secret: %w", err)
 }
 
-type generateCAFunc = func() ([]byte, *admissionregistration.KeyPair, error)
-type pemEncodeCertificateFunc = func(certificate []byte) ([]byte, error)
+func (r *Reconciler) fetchOrInitializePolicyServerSecret(ctx context.Context, policyServerName string, caSecret *corev1.Secret, generateCert generateCertFunc) (*corev1.Secret, error) {
+	policyServerSecret := corev1.Secret{}
+	err := r.Client.Get(
+		ctx,
+		client.ObjectKey{
+			Namespace: r.DeploymentsNamespace,
+			Name:      constants.PolicyServerSecretNamePrefix + policyServerName},
+		&policyServerSecret)
+	if err != nil && apierrors.IsNotFound(err) {
+		return r.buildPolicyServerSecret(policyServerName, caSecret, generateCert)
+	}
+	policyServerSecret.ResourceVersion = ""
```

Comment:
this line should be moved after the next error check

Reply:
yes, you are right. Thanks for finding this. For the other comments see my reply to the first one

Reply:
see next commit